sql-plugin/src/main/scala/com/nvidia/spark/rapids/DumpUtils.scala

```diff
@@ -0,0 +1,285 @@
+    }
+  }
+
+  private def dumpToParquetFileImpl(columnarBatch: ColumnarBatch, filePrefix: String): String = {
+    // generate path, retry if file exists
+    val path = genPath(filePrefix)
+
+    // construct schema according field types
+    val schema = genSchema(columnarBatch)
+
+    // write data, Note not close the column
+    val dumper = new GpuColumnBatchToParquetDumper(path, schema)
+    try {
+      dumper.writeBatch(columnarBatch)
+    } finally {
+      dumper.close()
+    }
+
+    path
+  }
+
+  private def genSchema(columnarBatch: ColumnarBatch): StructType = {
+    val fields = new Array[StructField](columnarBatch.numCols())
+    for (i <- 0 until columnarBatch.numCols()) {
+      // ,() are invalid in column name, replace these characters
+      val name = ("c" + i + "_" + columnarBatch.column(i).dataType())
+        .replace(',', '_')
+        .replace('(', '[')
+        .replace(')', ']')
+      fields(i) = StructField(name, columnarBatch.column(i).dataType())
+    }
+    StructType(fields)
+  }
+
+  private def genPath(filePrefix: String): String = {
+    var path = ""
+    val random = new Random
+    var succeeded = false
+    while (!succeeded) {
+      path = filePrefix + random.nextInt(Int.MaxValue) + ".parquet"
+      if (!new File(path).exists()) {
+        succeeded = true
+      }
+    }
+    path
+  }
+}
+
+// used for dump column batch to parquet
+class GpuColumnBatchToParquetDumper(
+    path: String,
+    schema: StructType)
+  extends ParquetDumper(path) {
+
+  override val tableWriter: TableWriter = {
+    val writeOptionBuilder = GpuParquetFileFormat
+      // avoid anything conversion, just dump as it is
+      .parquetWriterOptionsFromSchema(ParquetWriterOptions.builder(), schema, writeInt96 = false)
+      .withCompressionType(ParquetDumper.COMPRESS_TYPE)
+    Table.writeParquetChunked(writeOptionBuilder.build(), this)
+  }
+}
+
+// used for dump table to parquet
+class GpuTableToParquetDumper(
+    path: String,
+    table: Table)
+  extends ParquetDumper(path) {
+
+  override val tableWriter: TableWriter = {
+    // avoid anything conversion, just dump as it is
+    val builder = ParquetDumper.parquetWriterOptionsFromTable(ParquetWriterOptions.builder(), table)
+      .withCompressionType(ParquetDumper.COMPRESS_TYPE)
+    Table.writeParquetChunked(builder.build(), this)
+  }
+}
+
+// parquet dumper
+abstract class ParquetDumper(path: String) extends HostBufferConsumer with Arm {
+  private[this] val outputStream = new FileOutputStream(path)
+  private[this] val tempBuffer = new Array[Byte](128 * 1024)
+  private[this] val buffers = mutable.Queue[(HostMemoryBuffer, Long)]()
+
+  val tableWriter: TableWriter
+
+  override
+  def handleBuffer(buffer: HostMemoryBuffer, len: Long): Unit =
+    buffers += Tuple2(buffer, len)
+
+  def writeBufferedData(): Unit = {
+    ColumnarOutputWriter.writeBufferedData(buffers, tempBuffer, outputStream)
+  }
+
+  def writeBatch(batch: ColumnarBatch): Long = {
```

Comment:
This really should just be the following to reduce any kind of duplication in the code.

```
withResource(GpuColumnVector.from(batch)) { table =>
    writeTable(table)
}
```